modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaDLFolderMethodCheck.java

```diff
@@ -0,0 +1,120 @@
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaDLFolderMethodCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			Matcher matcher = _addFolderPattern.matcher(javaMethodContent);
+
+			while (matcher.find()) {
+				String methodCall = JavaSourceUtil.getMethodCall(
+					javaMethodContent, matcher.start());
+
+				String message = StringBundler.concat(
+					"Unable to format method addFolder from DLFolderService, ",
+					"DLFolderLocalService, DLFolderServiceUtil and ",
+					"DLFolderLocalServiceUtil. Fill the new parameter ",
+					"manually, see LPS-194001.");
+
+				String variableName = getVariableName(methodCall);
+
+				String newMethodCall = null;
+
+				List<String> parameterList = JavaSourceUtil.getParameterList(
+					methodCall);
+
+				if (variableName.contains("DLFolderLocalServiceUtil") ||
+					hasClassOrVariableName(
+						"DLFolderLocalService", newContent, newContent,
+						methodCall)) {
+
+					String[] parameterTypes = {
+						"long", "long", "long", "boolean", "long", "String",
+						"String", "boolean", "ServiceContext"
+					};
+
+					if (!hasValidParameters(
+							9, fileName, javaMethodContent, message,
+							parameterList, parameterTypes)) {
+
+						continue;
+					}
+
+					newMethodCall = JavaSourceUtil.addMethodNewParameters(
+						JavaSourceUtil.getIndent(methodCall), new int[] {0},
+						matcher.group(1), new String[] {"null"}, parameterList);
+
+					newContent = StringUtil.replace(
+						newContent, methodCall, newMethodCall);
+				}
+				else if (variableName.contains("DLFolderServiceUtil") ||
+						 hasClassOrVariableName(
+							 "DLFolderService", newContent, newContent,
+							 methodCall)) {
+
+					String[] parameterTypes = {
+						"long", "long", "boolean", "long", "String", "String",
+						"ServiceContext"
+					};
+
+					if (!hasValidParameters(
+							7, fileName, javaMethodContent, message,
+							parameterList, parameterTypes)) {
+
+						continue;
+					}
+
+					newMethodCall = JavaSourceUtil.addMethodNewParameters(
+						JavaSourceUtil.getIndent(methodCall), new int[] {0},
+						matcher.group(1), new String[] {"null"}, parameterList);
+
+					newContent = StringUtil.replace(
+						newContent, methodCall, newMethodCall);
+				}
+			}
+		}
+
+		return newContent;
+	}
+
+	private static final Pattern _addFolderPattern = Pattern.compile(
+		"(\\t*(|_dlFolderLocalService|DLFolderLocalServiceUtil|" +
+			"_dlFolderService|DLFolderServiceUtil)(\\.addFolder\\())");
```

Comment:
Hi @TamyrisTorres, please adjust the patterns and add a check of the type of the variable. Thanks

Reply:
resolved